contracts/modules/UpgradableModuleFactory.sol

```diff
@@ -0,0 +1,88 @@
+pragma solidity ^0.5.0;
+
+import "./ModuleFactory.sol";
+import "../proxy/OwnedUpgradeabilityProxy.sol";
+
+/**
+ * @title Factory for deploying upgradable modules
+ */
+contract UpgradableModuleFactory is ModuleFactory {
+
+    event LogicContractSet(string _version, address _logicContract, bytes _logicData);
+
+    struct LogicContract {
+        string version;
+        address logicContract;
+        bytes logicData;
+    }
+
+    // Mapping from version to logic contract
+    mapping (uint256 => LogicContract) logicContracts;
+
+    // Mapping from Security Token address to module version
+    mapping (address => uint256) moduleVersions;
+
+    // Mapping from Security Token address to module address
+    mapping (address => address) modules;
+
+    // Current version
+    uint256 public latestVersion;
+
+    /**
+     * @notice Constructor
+     * @param _setupCost Setup cost of the module
+     * @param _usageCost Usage cost of the module
+     * @param _logicContract Contract address that contains the logic related to `description`
+     * @param _polymathRegistry Address of the Polymath registry
+     */
+    constructor(
+        string memory _version,
+        uint256 _setupCost,
+        uint256 _usageCost,
+        address _logicContract,
+        address _polymathRegistry
+    )
+        public ModuleFactory(_setupCost, _usageCost, _polymathRegistry)
+    {
+        require(_logicContract != address(0), "Invalid address");
+        logicContracts[latestVersion].logicContract = _logicContract;
+        logicContracts[latestVersion].version = _version;
+    }
+
+    function setLogicContract(string calldata _version, address _logicContract, bytes calldata _logicData) external onlyOwner {
+        require(keccak256(abi.encodePacked(_version)) != keccak256(abi.encodePacked(logicContracts[latestVersion].version)), "Same version");
+        require(_logicContract != logicContracts[latestVersion].logicContract, "Same version");
+        require(_logicContract != address(0), "Invalid address");
+        latestVersion++;
```

Comment:
We are now essentially splitting version into this integer version and the string human readable version. I think we should drop the string version from modules altogether. I don't see it serving any real purpose.

Reply:
I'm treating the integer string (`latestVersion`) as more of an internal variable to index logic contract deployments. Will have a think about this in a later PR though.